paradox-nakadi-consumer-core/src/main/java/de/zalando/paradox/nakadi/consumer/core/http/handlers/AbstractEventsResponseBulkHandler.java

```diff
@@ -37,26 +37,26 @@ public void onResponse(final String content) {
 
             final List<T> batchEvents = optionalBatch.get().getEvents();
             if (null != batchEvents && !batchEvents.isEmpty()) {
-                handleEvents(cursor, batchEvents);
+                handleEvents(cursor, batchEvents, content);
             } else {
                 try {
                     log.info("Keep alive offset [{}]", cursor.getOffset());
                     coordinator.commit(cursor);
                 } catch (Throwable t) {
                     log.error("Handler error at cursor [{}]", cursor);
-                    coordinator.error(t, eventTypePartition);
+                    coordinator.error(t, eventTypePartition, cursor.getOffset(), content);
                 }
             }
         }
     }
 
-    private void handleEvents(final EventTypeCursor cursor, final List<T> events) {
+    private void handleEvents(final EventTypeCursor cursor, final List<T> events, final String content) {
         try {
             delegate.onEvent(cursor, events);
             coordinator.commit(cursor);
-        } catch (Throwable t) {
+        } catch (final Throwable t) {
             log.error("Handler error at cursor [{}]", cursor);
-            coordinator.error(t, eventTypePartition);
+            coordinator.error(t, eventTypePartition, cursor.getOffset(), content);
```

Comment:
[C] `AbstractEventsResponseBulkHandler.getEventBatch()` still returns `Optional<NakadiEventBatch<T>>` but the exception should be propagated.